scipy/stats/_morestats.py

```diff
@@ -3577,13 +3577,9 @@ def wilcoxon(x, y=None, zero_method="wilcox", correction=False,
       and ``method='exact'`` is preferred (at the cost of additional
       execution time).
 
-    - The default, ``method='auto'``, selects between the two: when
-      ``len(d) <= 50`` and there are no zeros and no ties, the exact method
-      is used; if the sample size is small and there are zeros or ties, the
-      p-value is computed using `permutation_test`;
-      otherwise, the approximate method is used. The p-value computed by
-      the permutation test is deterministic since it is only used if the
-      sample size is small enough to iterate over all possible outcomes.
+    - The default, ``method='auto'``, selects between the two:
+    ``method='exact'`` is used when ``len(d) <= 50``, and
+    ``method='asymptotic'`` is used otherwise.
```

Comment:
We have not yet defined "ties" or "zeros". Above, it specifies

> Assume that all elements of ``d`` are independent and identically distributed observations, and all are distinct and nonzero.

So let's document what happens under this assumption, then define ties and zeros and document how those affect the behavior.